storage-proofs/core/src/gadgets/insertion.rs

```diff
@@ -125,6 +308,11 @@ where
         }
     })?;
 
+    // Enforce (b - a) * cond = (b - c)
```

Comment:
I think that this comment is unnecessary and confuses the point. Annotating a quadratic constraint using algebraic manipulations implies (imo) that r1cs performs that algebra and/or that the linear terms are able to commute.

Reply:
I added it to clarify for the internal auditors, and I think at least the first line should stay (and is consistent with Bellman internal documentation). I can remove the algebra and replace it with a prose summary n the Bellman style.

I'm not sure what you mean by 'linear terms commute', but we can discuss that elsewhere.

@nicola If you do want the original algebra to remain, please say so — otherwise, I'll remove per @DrPeterVanNostrand's suggestion.

Reply:
The first line in the comment is good with me. My comment was meant express that the algebra was misleading because the `=`'s sign could be interpreted as equivalency of circuit wiring, not just arithmetic. The current comment is good with me.